spec/unit/file_serving/http_metadata_spec.rb

```diff
@@ -0,0 +1,70 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+require 'puppet/file_serving/http_metadata'
+require 'matchers/json'
+require 'net/http'
+require 'digest'
+
+describe Puppet::FileServing::HttpMetadata do
+  let(:foobar) { File.expand_path('/foo/bar') }
+
+  it "should be a subclass of Metadata" do
+    expect( described_class.superclass ).to be Puppet::FileServing::Metadata
+  end
+
+  describe "when initializing" do
```

Comment:
Should add a test when the server returns neither `Content-MD5` nor `Last-Modified`.